tdigest.go

```diff
@@ -116,30 +132,33 @@ func (t *TDigest) process() {
 		}
 		t.min = math.Min(t.min, t.processed[0].Mean)
 		t.max = math.Max(t.max, t.processed[t.processed.Len()-1].Mean)
-		t.updateCumulative()
 		t.unprocessed.Clear()
 	}
 }
 
 // Centroids returns a copy of processed centroids.
 // Useful when aggregating multiple t-digests.
 //
-// Pass in the CentroidList as the buffer to write into.
+// Centroids are appended to the passed CentroidList; if you're re-using a
+// buffer, be sure to pass cl[:0].
 func (t *TDigest) Centroids(cl CentroidList) CentroidList {
 	t.process()
 	return append(cl, t.processed...)
 }
 
 func (t *TDigest) Count() float64 {
 	t.process()
-	count := 0.0
-	for _, centroid := range t.processed {
-		count += centroid.Weight
-	}
-	return count
+	return t.processedWeight
```

Comment:
Is the idea here that any time `t.process()` is called we have also called `t.updateCummulative()` so that we know we can just return `t.processedWeight` since it has been correctly updated? 

In other words why is it now safe to return ` t.processedWeight` instead of computing it for each call to count? Can you add that explanation as a comment here so that future changes know what invariants need to be held?

Reply:
That's the basic idea except that `t.processedWeight` is updated by `t.process()` directly, not `t.updateCummulative()`. I'll add a comment.